service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java

```diff
@@ -0,0 +1,265 @@
+ * The outcome is then returned to the client in the HTTP response.
+ */
+public abstract class AbstractCredentialsManagementHttpEndpoint extends AbstractHttpEndpoint<ServiceConfigProperties> {
+
+
+    private static final String SPAN_NAME_GET_CREDENTIALS = "get Credentials from management API";
+    private static final String SPAN_NAME_UPDATE_CREDENTIALS = "update Credentials from management API";
+
+    /**
+     * Creates an endpoint for a Vertx instance.
+     *
+     * @param vertx The Vertx instance to use.
+     * @throws NullPointerException if vertx is {@code null};
+     */
+    @Autowired
+    public AbstractCredentialsManagementHttpEndpoint(final Vertx vertx) {
+        super(Objects.requireNonNull(vertx));
+    }
+
+    /**
+     * Returns an empty String as this implementation does not use event bus.
+     */
+    @Override
+    protected String getEventBusAddress() {
+        return "";
+    }
+
+    @Override
+    public String getName() {
+        return String.format("%s/%s",
+                RegistryManagementConstants.API_VERSION,
+                RegistryManagementConstants.CREDENTIALS_HTTP_ENDPOINT);
+    }
+
+    @Override
+    public void addRoutes(final Router router) {
+
+        final String pathWithTenantAndDeviceId = String.format("/%s/:%s/:%s",
+                getName(), PARAM_TENANT_ID, PARAM_DEVICE_ID);
+
+
+        // Add CORS handler
+        router.route(pathWithTenantAndDeviceId).handler(createCorsHandler(config.getCorsAllowedOrigin(), EnumSet.of(HttpMethod.GET, HttpMethod.PUT)));
+
+        final BodyHandler bodyHandler = BodyHandler.create();
+        bodyHandler.setBodyLimit(config.getMaxPayloadSize());
+
+        // get all credentials for a given device
+        router.get(pathWithTenantAndDeviceId).handler(this::getCredentialsForDevice);
+
+        // set credentials for a given device
+        router.put(pathWithTenantAndDeviceId).handler(bodyHandler);
+        router.put(pathWithTenantAndDeviceId).handler(this::extractRequiredJsonArrayPayload);
+        router.put(pathWithTenantAndDeviceId).handler(this::extractIfMatchVersionParam);
+        router.put(pathWithTenantAndDeviceId).handler(this::updateCredentials);
+    }
+
+
+    /**
+     * The service to forward requests to.
+     *
+     * @return The service to bind to, must never return {@code null}.
+     */
+    protected abstract CredentialsManagementService getService();
+
+    private void updateCredentials(final RoutingContext ctx) {
+
+        final Span span = Util.newChildSpan(SPAN_NAME_UPDATE_CREDENTIALS, TracingHandler.serverSpanContext(ctx), tracer,
+                getClass().getSimpleName());
+
+        final JsonArray credentials = ctx.get(KEY_REQUEST_BODY);
+
+        final String deviceId = getDeviceIdParam(ctx);
+        final Optional<String> resourceVersion = Optional.ofNullable(ctx.get(KEY_RESOURCE_VERSION));
+        final String tenantId = getTenantParam(ctx);
+
+        if (tenantId == null) {
```

Comment:
Take a look at how `Util.getRequestParam()` is used in tenant implementation. We should be consistent here and change device endpoint as well.